thesis-api-v2/src/main/java/pl/edu/agh/quizzesthesis/api/controller/TeacherQuestionsSetController.java

```diff
@@ -0,0 +1,49 @@
+package pl.edu.agh.quizzesthesis.api.controller;
+
+import lombok.AllArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+import pl.edu.agh.quizzesthesis.api.dto.*;
+import pl.edu.agh.quizzesthesis.business.Current;
+import pl.edu.agh.quizzesthesis.business.UserAuthDetails;
+import pl.edu.agh.quizzesthesis.business.service.TeacherQuestionsSetService;
+
+import java.util.List;
+
+import static pl.edu.agh.quizzesthesis.App.API_URL_PREFIX;
+
+@RestController
+@RequestMapping(API_URL_PREFIX + "/questions-set")
+@AllArgsConstructor
+public class TeacherQuestionsSetController {
+    private final TeacherQuestionsSetService teacherQuestionsSetService;
+
+    @GetMapping
+    public QuestionsSetResponse getQuestionsSet(@RequestParam(name = "key") String questionsSetKey) {
+        return teacherQuestionsSetService.getQuestionsSet(questionsSetKey);
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public QuestionsSetKeyResponse createQuestionsSet(@Current UserAuthDetails userAuthDetails,
+                                                      @RequestBody QuestionsSetsRequest request) {
+        return teacherQuestionsSetService.createQuestionsSet(userAuthDetails, request);
+    }
+
+    @DeleteMapping
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void deleteQuestionsSet(@Current UserAuthDetails userAuthDetails, @RequestBody QuestionsSetKeyRequest request) {
+        teacherQuestionsSetService.deleteQuestionsSet(userAuthDetails, request.questionsSetKey());
+    }
+
+    @GetMapping("/list")
```

Comment:
`/list` in endpoint is not a RESTful way of creating URLs. Perhaps you could use "/" here and "/keys/{key}" in `getQuestionsSet` instead of `RequestParam`?

Reply:
Changed as suggested.

Reply:
Sorry, my mistake. I gave you wrong suggestion about that enpoint URL. `"/keys/"` part is unecessary. It should be just `"/{key}"`, because key is an ID of the resource (REST convention `"/resources-name/{id}"`). Once again, sorry. I didn't think it through before writing the previous comment.

Reply:
Changed to `/{key}`.